guide.adoc

```diff
@@ -0,0 +1,596 @@
+
+== What you'll build
+
+The application you will be building exposes a very simple greeting web service accepting `GET` HTTP requests to
+`/api/greeting`, responding with a JSON message in the form `{"content":"Hello, World!"}`. The
+message can be customized by passing the `name` query parameter to the request as in
+`http://localhost:8080/api/greeting?name=John` which would result in the following response: `{"content":"Hello,
+John!"}`.
+Additionally, a very simple front-end is provided using HTML and jQuery to interact with the greeting endpoint
+from a more user-friendly interface than query the service using https://curl.haxx.se/[`cURL`].
+
+NOTE: While we will go over the implementation, we will focus mostly on the specifics needed to get your application running on
+OpenShift as opposed to detailing all the Spring Boot-specific implementation.
+
+== What you'll need
+
+* Java 8
+* Access to an OpenShift cluster whether locally via https://www.openshift.org/minishift/[minishift] or using the different
+  flavors of https://www.openshift.com/products[OpenShift products]
+
+
+== Endpoint
+
+The application is composed of a RESTful service which code can be found in the `src/main/java/io/openshift/booster/service`
+directory. It is split in two classes: `GreetingEndpoint`, which is implements the endpoint itself, and `Greeting` which is a
+simple class representing the payload sent back to users of the endpoint.
+
+Let's look at the `Greeting` class first, which is pretty simple:
+```java
+public class Greeting {
+
+    public static final String FORMAT = "Hello, %s!"; #<1>
+
+    private final String content; #<2>
+
+    public Greeting() {
+        this.content = null;
+    }
+
+    public Greeting(String content) {
+        this.content = content;
+    }
+
+    public String getContent() {
+        return content;
+    }
+}
+```
+<1> `FORMAT` constant that can be used by clients of the class to create greeting messages in the expected format
+<2> `content` field which will be used to populate our JSON payload
+This `Greeting` class will be automatically marshalled by Jackson using the accessor.
+
+Let's now look at the `GreetingEndpoint` class, short and sweet but packing quite a punch, thanks to annotations:
+```java
+import javax.ws.rs.DefaultValue; # <1>
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+
+import org.springframework.stereotype.Component;
+
+@Path("/greeting") # <2>
+@Component # <3>
+public class GreetingEndpoint {
+    @GET # <4>
+    @Produces("application/json") # <5>
+    public Greeting greeting(@QueryParam("name") @DefaultValue("World") String name) { #<6>
+        final String message = String.format(Greeting.FORMAT, name); #<7>
+        return new Greeting(message); #<8>
+    }
+}
+```
+<1> Standard JAX-RS annotation imports
+<2> Specify that this class is a JAX-RS root resource and that the endpoint will answer requests on `/greeting`
```

Comment:
answer -> serve ?

Reply:
Not sure here. It serves responses but answers requests, no?

Reply:
That's true, it's just that I haven't seen the term used much in the literature